File: app/src/main/java/me/niccorder/phunware/data/repository/WeatherRepository.java
package me.niccorder.phunware.data.repository;

import io.reactivex.Observable;
import io.reactivex.annotations.NonNull;
import me.niccorder.phunware.model.Forecast;
import me.niccorder.phunware.model.Location;

/**
 * A repository contract for interacting with all information for weather.
 */
public interface WeatherRepository {

  /**
   * @param location of the forcast to be retrieved.
   * @return an {@link Observable} which emits the {@link Forecast} for the given {@link Location}.
   */
  Observable<Forecast> getForecast(@NonNull final Location location);
}

File: data/src/main/java/me/niccorder/phunware/data/WeatherRepository.kt
package me.niccorder.phunware.data

import io.reactivex.Observable
import me.niccorder.phunware.model.Forecast
import me.niccorder.phunware.model.Location

/**
 * A repository contract for interacting with all information for weather.
 */
interface WeatherRepository {

  /**
   * @param location of the forcast to be retrieved.
   * @return an [Observable] which emits the [Forecast] for the given [Location].
   */
  fun getForecast(location: Location): Observable<Forecast>
}
